src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/ICustomTypeMarshallingStrategy.cs

```diff
@@ -97,6 +126,9 @@ public IEnumerable<StatementSyntax> GeneratePinnedMarshalStatements(TypePosition
 
         public IEnumerable<StatementSyntax> GenerateUnmarshalStatements(TypePositionInfo info, StubCodeContext context)
         {
+            if (!_shape.HasFlag(MarshallerShape.ToManaged))
+                yield break;
```

Comment:
Any particular reason to do the checks in the single "stateless" implementation of the interface instead of following the design I did with v1 of having a set of nested instances with each implementation providing the support for one feature?

Reply:
I was trying to hit somewhere in the middle - putting most basic stateless support into the one implementation and separating the ones that require more information (like caller-allocated buffer) or would be reused (like static pinnable). I think the nested instances can get a bit hard to track when it gets deep (or just browsing code now that we would also have stateless and stateful versions for each feature), so I pushed a bit more into the root stateless implementation (especially since stateless generation is relatively simple).